node/test/util/rm.js

```diff
@@ -0,0 +1,396 @@
+                initial: "x=S:C2-1 x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                path: "x/y",
+                recursive: false,
+                fails: true
+            },
+            "rm two clean files by removing their containing dir": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                path: "x/y",
+                recursive: true,
+                output: {
+                    isRootLevel: true,
+                    pathFromRoot: "x/y"
+                }
+            },
+            "rm two clean files by removing their grandparent dir": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;Bmaster=2",
+                path: "x",
+                recursive: true,
+                output: {
+                    isRootLevel: true,
+                    pathFromRoot: "x"
+                }
+            },
+            "rm files by removing their containing dir with untracked in wd": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                path: "x",
+                recursive: true,
+                output: {
+                    isRootLevel: true,
+                    pathFromRoot: "x"
+                }
+            },
+            "rm a non-existent thing": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;W x/q/z=bar;Bmaster=2",
+                path: "z",
+                fails: true
+            },
+            "rm a submodule": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;C3-1 d=S/baz.git:2;" +
+                    "Bmaster=3;Bsub=2",
+                path: "d",
+                output: {
+                    isRootLevel: true,
+                    pathFromRoot: "d"
+                }
+            },
+            // Case 1
+            "rm a non-existent thing from a submodule": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;C3-1 d=S/baz.git:2;" +
+                    "Bmaster=3;Bsub=2",
+                path: "d/f",
+                fails: true
+            },
+            "rm a file from a closed submodule": {
+                initial: "x=S:C2-1 x/y/a=foo,x/y/z=foo;C3-1 d=S/baz.git:2;" +
+                    "Bmaster=3;Bsub=2",
+                path: "d/x/y/z",
+                fails: true
+            },
+            "rm a file from an open submodule": {
+                initial: "sub=S:C2-1 x/y/a=foo,x/y/z=foo;Bmaster=2|" +
+                    "x=S:C3-1 d=Ssub:2;" +
+                    "Bmaster=3;Od",
+                path: "d/x/y/z",
+                output: {
+                    isRootLevel: false,
+                    pathFromRoot: "d",
+                    pathInSubmodule: "x/y/z"
+                }
+            },
+            "rm a nonexistent dir from an open submodule": {
+                initial: "sub=S:C2-1 x/y/a=foo,x/y/z=foo;Bmaster=2|" +
+                    "x=S:C3-1 d=Ssub:2;" +
+                    "Bmaster=3;Od",
+                path: "d/q",
```

Comment:
The parser is more graceful about line breaks now, so you can write this using the "`" string syntax that allows for embedded eols rather than string concatenation.